scheduler/src/cook/mesos/util.clj

```diff
@@ -364,25 +422,31 @@
                                    (-> 1 t/hours t/in-millis)))
           expanded-end (Date. (+ (.getTime end)
                                  (-> 1 t/hours t/in-millis)))
+          entid-start (d/entid-at db :db.part/user expanded-start)
+          entid-end (d/entid-at db :db.part/user expanded-end)
           default-pool? (pool/default-pool? pool-name)
           pool-name' (or pool-name pool/nil-pool)
+          job-user-entid (d/entid db :job/user)
+          start-ms (.getTime start)
+          end-ms (.getTime end)
           jobs
-          (->> (d/seek-datoms db :avet :job/user user (d/entid-at db :db.part/user expanded-start))
-               (take-while #(and (< (.e %) (d/entid-at db :db.part/user expanded-end))
-                                 (= (.a %) (d/entid db :job/user))
-                                 (= (.v %) user)))
-               (map #(.e %))
+          (->> (d/seek-datoms db :avet :job/user user entid-start)
+               (take-while #(and (< (:e %) entid-end)
+                                 (= (:a %) job-user-entid)
+                                 (= (:v %) user)))
+               (map :e)
                (map (partial d/entity db))
-               (filter #(<= (.getTime start) (.getTime (:job/submit-time %))))
-               (filter #(< (.getTime (:job/submit-time %)) (.getTime end)))
+               (filter #(job-submit-time-between % start-ms end-ms))
                (filter #(= :job.state/completed (:job/state %)))
                (filter #(pool/check-pool-for-listing % :job/pool pool-name' default-pool?)))]
       (->>
         (cond->> jobs
                  (not include-custom-executor?) (filter #(false? (:job/custom-executor %)))
                  (instance-states state) (filter #(= state (job-ent->state %)))
                  name-filter-fn (filter #(name-filter-fn (:job/name %))))
-        (take limit)))))
+        ; No need to sort. We're traversing in entity_id order, so in time order.
+        (take limit)
+        doall))))
```

Comment:
The function above mentions in the docstring that it's returning lazy output. Should we mention in this function's docstring that it's returning non-lazy output?

Reply:
Honestly, no. I think non-lazy should be the usual assumption. Are we thinking of adding 'not lazy' to every function thats not lazy? 

I can strip it off the other function, but I don't think it harms anything.

Reply:
> Are we thinking of adding 'not lazy' to every function thats not lazy?

I'm not advocating for that.

The comment was mostly about consistency, since the two functions are so closely related. If you think it's clear now, I'm good with it.